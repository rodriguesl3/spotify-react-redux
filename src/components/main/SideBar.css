.sideBarContent {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  width: 80%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: reverse;
      -ms-flex-direction: row-reverse;
          flex-direction: row-reverse;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin-left: 10px;
  background-color: #23395B;
}

.imageProfile {
  background-size: cover;
  width: 100%;
  max-width: 150px;
  border-radius: 500px;
  padding: 20px;
}

.userName {
  justify-items: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  font-size: 30px;
  -webkit-transition: 0.7s;
  transition: 0.7s;
}
/*# sourceMappingURL=SideBar.css.map */