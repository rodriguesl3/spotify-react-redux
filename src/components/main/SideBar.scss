

.sideBarContent {
    transition: 0.5s;
    border-right: 2px solid black;
    width: 40%;
    display: flex;
    flex-direction: row-reverse;
    justify-content: center;
    align-items: center;
}

.sideBarContent:hover {
    width: 80%;
}

.sideBarContent:hover .userName {
   
    font-size: 30px;
}

.imageProfile {
    background-size: cover;
    width: 63%;
    max-width: 150px;
    border-radius: 500px;
    padding: 20px;
}

.userName {
    display: flex;
    justify-content: center;
    align-content: center;
    font-size: 15px;
    transition: 0.7s;
}