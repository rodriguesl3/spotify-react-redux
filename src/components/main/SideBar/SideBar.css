.sideBarContent {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  width: 80%;
  height: 100vh;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0;
  background-color: gray;
  position: absolute;
  z-index: 1;
  left: -269px;
  color: white;
}

.sideBarContent-full {
  width: 110%;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  height: 100vh;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0;
  background-color: gray;
  position: absolute;
  z-index: 1;
  left: 0px;
  color: white;
}

.sideBarContent-full .closeButton {
  height: 50px;
  margin-top: 25px;
}

.imageProfile {
  border-radius: 500px;
  margin: 10px 20px 10px 10%;
}

.userName {
  justify-items: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  font-size: 30px;
  -webkit-transition: 0.7s;
  transition: 0.7s;
  margin-left: 20px;
}
/*# sourceMappingURL=SideBar.css.map */