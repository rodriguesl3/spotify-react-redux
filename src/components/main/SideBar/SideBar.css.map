{
    "version": 3,
    "mappings": "AAEA,AAAA,eAAe,CAAC;EACZ,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,CAAC;EACd,gBAAgB,ECDJ,IAAI;EDEhB,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,CAAC;EACV,IAAI,EAAE,MAAM;EACZ,KAAK,ECHW,KAAK;CDKxB;;AAED,AAAA,oBAAoB,CAAC;EACjB,KAAK,EAAE,IAAI;EACX,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,CAAC;EACd,gBAAgB,ECfJ,IAAI;EDgBhB,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,CAAC;EACV,IAAI,EAAE,GAAG;EACT,KAAK,ECjBW,KAAK;CDuBxB;;AAhBD,AAYI,oBAZgB,CAYhB,YAAY,CAAC;EACT,MAAM,EAAE,IAAI;EACZ,UAAU,EAAE,IAAI;CACnB;;AAIL,AAAA,aAAa,CAAC;EACV,aAAa,EAAE,KAAK;EACpB,MAAM,EAAE,kBAAkB;CAC7B;;AAED,AAAA,SAAS,CAAC;EACN,aAAa,EAAE,MAAM;EACrB,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,IAAI;EAChB,WAAW,EAAE,IAAI;CACpB",
    "sources": [
        "SideBar.scss",
        "../../../base.scss"
    ],
    "names": [],
    "file": "SideBar.css"
}