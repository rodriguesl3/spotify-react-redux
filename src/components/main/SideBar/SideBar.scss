@import "../../../base.scss";

.sideBarContent {
    transition: 0.5s;
    width: 80%;
    height: 100vh;
    display: flex;
    margin-left: 0;
    background-color: $primary-color;
    position: absolute;
    z-index: 1;
    left: -269px;
    color: $alternative-color;

}

.sideBarContent-full {
    width: 80%;
    transition: 0.5s;
    width: 80%;
    height: 100vh;
    display: flex;
    margin-left: 0;
    background-color: $primary-color;
    position: absolute;
    z-index: 1;
    left: 0px;
    color: $alternative-color;

    .closeButton {
        height: 50px;
        margin-top: 25px;
    }
}


.imageProfile {
    border-radius: 500px;
    margin: 10px 20px 10px 10%;
}

.userName {
    justify-items: center;
    align-items: center;
    font-size: 30px;
    transition: 0.7s;
    margin-left: 20px;
}