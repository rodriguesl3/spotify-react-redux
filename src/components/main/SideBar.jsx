import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './SideBar.scss';

const imageUrl = "https://www.google.com/url?sa=i&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwjY0q2jsMffAhXCvJAKHXo0B68QjRx6BAgBEAU&url=https%3A%2F%2Fpt.wikipedia.org%2Fwiki%2FAnonymous&psig=AOvVaw0LvWcw8Dbz2hlIjV2raw_r&ust=1546253220121470"


export const SideBar = (props) => {
    const { spotifyUserInfo, onGetFollowingArtist } = props;

    const GetFollowingArtistHandle = () => {
        onGetFollowingArtist();
    }

    return (
        <div className="container">
            <div className="row sideBarContent">
                <img src={(spotifyUserInfo)
                    ? (spotifyUserInfo.images[0].url)
                    : (imageUrl)} alt=""
                    className="col-sm-5 imageProfile"
                />

                <div className="col-sm-12 userName">
                    {spotifyUserInfo.display_name}
                </div>
                <div className="row sideBarList">
                    <div className="col-sm-6" onClick={GetFollowingArtistHandle}>Artists Following</div>
                </div>
            </div>
        </div>
    )
}

// SideBar.propTypes = {
//     spotifyUserInfo: PropTypes.object.isRequired,
// }