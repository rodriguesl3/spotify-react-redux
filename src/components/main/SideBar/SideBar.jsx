/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
// eslint-disable-next-line react/no-deprecated
import React from 'react';
import { MdClose } from 'react-icons/md/index';

import './SideBar.scss';
import { CustomCard } from '../../atoms/CustomCard/CustomCard';
import UserListen from '../../atoms/UserListen/UserListen';
import { imageUrl } from '../../../constants/index';


// eslint-disable-next-line import/prefer-default-export
const SideBar = ({
  spotifyUserInfo,
  showSideBar,
  onShowSideBar,
  userListening
}) => {
  const hideSideBarHandle = () => {
    onShowSideBar(false);
  };

  return (
    <div className={`row ${!showSideBar ? 'sideBarContent' : 'sideBarContent-full'}`}>
      <div className="col-12 userName">
        <button type="button" className="closeButton col-2 offset-9 " onClick={hideSideBarHandle}>
          <MdClose />
        </button>
        <img src={(spotifyUserInfo) ? (spotifyUserInfo.images[0].url) : (imageUrl)} alt="" className="imageProfile" />
        <p>{(spotifyUserInfo) && spotifyUserInfo.display_name}</p>
      </div>
      <div className="col-12">
        <CustomCard
          image={(userListening ? userListening.item.album.images[1].url : imageUrl)}
        >
          <UserListen userListening={userListening} />
        </CustomCard>
      </div>
    </div>
  );
};

export default SideBar;
