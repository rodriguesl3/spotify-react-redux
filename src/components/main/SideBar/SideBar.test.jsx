/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from './SideBar';
import { userInfo } from '../UserInfo/__mocks__/objectsMocks';

test('SideBar snaphot test', () => {
  const component = renderer.create(<SideBar showSideBar="true" spotifyUserInfo={userInfo.data} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
