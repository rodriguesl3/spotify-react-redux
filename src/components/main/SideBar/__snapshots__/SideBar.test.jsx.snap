// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar event test 1`] = `
<div
  className="row sideBarContent-full"
>
  <div
    className="col-12 userName"
  >
    <button
      className="closeButton col-2 offset-9 "
      onClick={[Function]}
      type="button"
    >
      <MdClose />
    </button>
    <img
      alt=""
      className="imageProfile"
      src="https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=1084373748241702&height=200&width=200&ext=1551600776&hash=AeTTgzEoEZf2Z_uj"
    />
    <p>
      Lucas Rodrigues
    </p>
  </div>
  <div
    className="col-12"
  >
    <CustomCard
      image="https://i.scdn.co/image/d517611a9aef994460eb73e5d38d59149e6f3fef"
    >
      <LoaderClassHOC
        userListening={
          Object {
            "currently_playing_type": "track",
            "is_playing": true,
            "item": Object {
              "album": Object {
                "album_type": "album",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "https://open.spotify.com/artist/4WN5naL3ofxrVBgFpguzKo",
                    },
                    "href": "https://api.spotify.com/v1/artists/4WN5naL3ofxrVBgFpguzKo",
                    "id": "4WN5naL3ofxrVBgFpguzKo",
                    "name": "Rudimental",
                    "type": "artist",
                    "uri": "spotify:artist:4WN5naL3ofxrVBgFpguzKo",
                  },
                ],
                "available_markets": Array [
                  "AD",
                  "AE",
                  "AR",
                  "AT",
                  "AU",
                  "BE",
                  "BG",
                  "BH",
                  "BO",
                  "BR",
                  "CA",
                  "CH",
                  "CL",
                  "CO",
                  "CR",
                  "CY",
                  "CZ",
                  "DE",
                  "DK",
                  "DO",
                  "DZ",
                  "EC",
                  "EE",
                  "EG",
                  "ES",
                  "FI",
                  "FR",
                  "GB",
                  "GR",
                  "GT",
                  "HK",
                  "HN",
                  "HU",
                  "ID",
                  "IE",
                  "IL",
                  "IS",
                  "IT",
                  "JO",
                  "JP",
                  "KW",
                  "LB",
                  "LI",
                  "LT",
                  "LU",
                  "LV",
                  "MA",
                  "MC",
                  "MT",
                  "MX",
                  "MY",
                  "NI",
                  "NL",
                  "NO",
                  "NZ",
                  "OM",
                  "PA",
                  "PE",
                  "PH",
                  "PL",
                  "PS",
                  "PT",
                  "PY",
                  "QA",
                  "RO",
                  "SA",
                  "SE",
                  "SG",
                  "SK",
                  "SV",
                  "TH",
                  "TN",
                  "TR",
                  "TW",
                  "US",
                  "UY",
                  "VN",
                  "ZA",
                ],
                "external_urls": Object {
                  "spotify": "https://open.spotify.com/album/0ORqx9M6ak7kqDwYUBwvdK",
                },
                "href": "https://api.spotify.com/v1/albums/0ORqx9M6ak7kqDwYUBwvdK",
                "id": "0ORqx9M6ak7kqDwYUBwvdK",
                "images": Array [
                  Object {
                    "height": 640,
                    "url": "https://i.scdn.co/image/1a6a17e359b8a39919c541505cd681aef4b7b294",
                    "width": 640,
                  },
                  Object {
                    "height": 300,
                    "url": "https://i.scdn.co/image/d517611a9aef994460eb73e5d38d59149e6f3fef",
                    "width": 300,
                  },
                  Object {
                    "height": 64,
                    "url": "https://i.scdn.co/image/06e2a532cd47b9a3c90f75e281cb382088d197a4",
                    "width": 64,
                  },
                ],
                "name": "Toast to our Differences (Deluxe)",
                "release_date": "2019-01-25",
                "release_date_precision": "day",
                "total_tracks": 16,
                "type": "album",
                "uri": "spotify:album:0ORqx9M6ak7kqDwYUBwvdK",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "https://open.spotify.com/artist/4WN5naL3ofxrVBgFpguzKo",
                  },
                  "href": "https://api.spotify.com/v1/artists/4WN5naL3ofxrVBgFpguzKo",
                  "id": "4WN5naL3ofxrVBgFpguzKo",
                  "name": "Rudimental",
                  "type": "artist",
                  "uri": "spotify:artist:4WN5naL3ofxrVBgFpguzKo",
                },
                Object {
                  "external_urls": Object {
                    "spotify": "https://open.spotify.com/artist/4ScCswdRlyA23odg9thgIO",
                  },
                  "href": "https://api.spotify.com/v1/artists/4ScCswdRlyA23odg9thgIO",
                  "id": "4ScCswdRlyA23odg9thgIO",
                  "name": "Jess Glynne",
                  "type": "artist",
                  "uri": "spotify:artist:4ScCswdRlyA23odg9thgIO",
                },
                Object {
                  "external_urls": Object {
                    "spotify": "https://open.spotify.com/artist/7LHLq2wayvWBPTwk6p9zZz",
                  },
                  "href": "https://api.spotify.com/v1/artists/7LHLq2wayvWBPTwk6p9zZz",
                  "id": "7LHLq2wayvWBPTwk6p9zZz",
                  "name": "Chronixx",
                  "type": "artist",
                  "uri": "spotify:artist:7LHLq2wayvWBPTwk6p9zZz",
                },
              ],
              "available_markets": Array [
                "AD",
                "AE",
                "AR",
                "AT",
                "AU",
                "BE",
                "BG",
                "BH",
                "BO",
                "BR",
                "CA",
                "CH",
                "CL",
                "CO",
                "CR",
                "CY",
                "CZ",
                "DE",
                "DK",
                "DO",
                "DZ",
                "EC",
                "EE",
                "EG",
                "ES",
                "FI",
                "FR",
                "GB",
                "GR",
                "GT",
                "HK",
                "HN",
                "HU",
                "ID",
                "IE",
                "IL",
                "IS",
                "IT",
                "JO",
                "JP",
                "KW",
                "LB",
                "LI",
                "LT",
                "LU",
                "LV",
                "MA",
                "MC",
                "MT",
                "MX",
                "MY",
                "NI",
                "NL",
                "NO",
                "NZ",
                "OM",
                "PA",
                "PE",
                "PH",
                "PL",
                "PS",
                "PT",
                "PY",
                "QA",
                "RO",
                "SA",
                "SE",
                "SG",
                "SK",
                "SV",
                "TH",
                "TN",
                "TR",
                "TW",
                "US",
                "UY",
                "VN",
                "ZA",
              ],
              "disc_number": 1,
              "duration_ms": 268653,
              "explicit": false,
              "external_ids": Object {
                "isrc": "GBAHS1800369",
              },
              "external_urls": Object {
                "spotify": "https://open.spotify.com/track/5gHm37FoBAppgJjGI5EA0w",
              },
              "href": "https://api.spotify.com/v1/tracks/5gHm37FoBAppgJjGI5EA0w",
              "id": "5gHm37FoBAppgJjGI5EA0w",
              "is_local": false,
              "name": "Dark Clouds (feat. Jess Glynne & Chronixx)",
              "popularity": 62,
              "preview_url": "https://p.scdn.co/mp3-preview/f40b823251dac30a4e5122e253f93903bd2a1c2c?cid=dc6af13b21094058a9b5e104eccee32e",
              "track_number": 3,
              "type": "track",
              "uri": "spotify:track:5gHm37FoBAppgJjGI5EA0w",
            },
          }
        }
      />
    </CustomCard>
  </div>
</div>
`;

exports[`SideBar snaphot test 1`] = `
<div
  className="row sideBarContent-full"
>
  <div
    className="col-12 userName"
  >
    <button
      className="closeButton col-2 offset-9 "
      onClick={[Function]}
      type="button"
    >
      <MdClose />
    </button>
    <img
      alt=""
      className="imageProfile"
      src="https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=1084373748241702&height=200&width=200&ext=1551600776&hash=AeTTgzEoEZf2Z_uj"
    />
    <p>
      Lucas Rodrigues
    </p>
  </div>
  <div
    className="col-12"
  >
    <CustomCard
      image="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a6/Anonymous_emblem.svg/800px-Anonymous_emblem.svg.png"
    >
      <LoaderClassHOC />
    </CustomCard>
  </div>
</div>
`;
